sql/moz-fx-data-shared-prod/telemetry_derived/active_users_aggregates_device_v1/query.sql

```diff
@@ -19,6 +19,16 @@ SELECT
   COUNT(DISTINCT IF(days_since_seen = 0, client_id, NULL)) AS dau,
   COUNT(DISTINCT IF(days_since_seen < 7, client_id, NULL)) AS wau,
   COUNT(DISTINCT client_id) AS mau,
+  COUNT(
+    DISTINCT IF(
+      days_since_seen = 0
+      AND normalized_app_name = 'Firefox Desktop'
+      AND active_hours_sum > 0
+      AND uri_count > 0,
+      client_id,
+      NULL
+    )
+  ) AS qdau,
```

Comment:
Nothing to do now, but it feels a bit fragile to me that this logic is defined in three different places:
- sql/moz-fx-data-shared-prod/telemetry_derived/active_users_aggregates_attribution_v1/query.sql
- sql/moz-fx-data-shared-prod/telemetry_derived/active_users_aggregates_device_v1/query.sql
- sql/moz-fx-data-shared-prod/telemetry_derived/active_users_aggregates_v1/query.sql

I'm still learning my way around this repo, but is there any chance that future work could consolidate this code in a single place, possibly upstream?

Reply:
That's the plan for a future version yes, for all the metrics ideally e.g. searches, ad clicks, active users. Using [metric hub](https://docs.google.com/presentation/d/1sXyekWMhYfXAwvEsCT5Q4Cjnu5owQ8LJ-VGyX7gGwCk/edit#slide=id.g82761e80df_0_1948) is the current proposal for this consolidation.